src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -30,38 +30,40 @@ public ArrayListProductDao(){
     }
     @Override
     public Product getProduct(Long id) {
-        return result.stream().filter(x-> x.getId()==id).findFirst().get();
-    }
-    
-    @Override
-    public boolean getProd(Long id){
-        return result.stream().anyMatch(x->x.getId()==id);
+        result.stream().filter(x-> x.getId().equals(id)).findFirst().
+                orElseThrow(NullPointerException::new);
+        
+        return result.stream().
+                filter(x-> x.getId().equals(id)).
+                findFirst().get();
     }
     
     @Override
     public List<Product> findProducts() {
-        List<Product> list = result.stream().filter(x-> x.getPrice()!=null && x.getStock()>0).collect(Collectors.toCollection(ArrayList::new));
+        List<Product> list = result.stream().
+                filter(x-> x.getPrice()!=null && x.getStock()>0).
+                collect(Collectors.toCollection(ArrayList::new));
         return list;
     }
 
     @Override
     public void save(Product product) {
-        if(this.result.stream().anyMatch(x-> x.getId()==product.getId())){
-            throw new IllegalArgumentException("Object's id isn't unic");
+        if(this.result.stream().anyMatch(x-> x.getId().equals(product.getId()))){
+            
+            this.result.stream().
+                filter(x-> x.getId().equals(product.getId())).
+                    findFirst().map(x-> this.result.set(this.result.indexOf(x), product));
         }
```

Comment:
The wrong indentation makes code difficult to read. Please compare:

this.result.stream().
                filter(x-> x.getId().equals(product.getId())).
                    findFirst().map(x-> this.result.set(this.result.indexOf(x), product));

AND

this.result.stream()
        .filter(x-> x.getId().equals(product.getId()))
        .findFirst()
        .map(x-> this.result.set(this.result.indexOf(x), product));

Please have a look: https://marcin-chwedczuk.github.io/java-streams-best-practices